libpod/container.go

```diff
@@ -1062,3 +1067,16 @@ func networkDisabled(c *Container) (bool, error) {
 	}
 	return false, nil
 }
+
+// State returns containerstate struct
+func (c *Container) ContainerState() (*ContainerState, error) {
+	if !c.batched {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return nil, err
+		}
+	}
+	return c.state, nil
```

Comment:
Make a copy via Deepcopier first - just handing out a pointer is risky, could get changes into the database

Reply:
correction ,fixed